app/src/main/java/org/jellyfin/mobile/AppPreferences.kt

```diff
@@ -52,6 +54,28 @@ class AppPreferences(context: Context) {
             }
         }
 
+    var downloadLocation: String
+        get() {
+            val defaultStorage = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).absolutePath
+            val savedStorage = sharedPreferences.getString(Constants.PREF_DOWNLOAD_LOCATION, null) ?: defaultStorage
+            return with(File(savedStorage)) {
+                parentFile?.run {
+                    if (exists() && isDirectory) return@with savedStorage
+                }
+                sharedPreferences.edit { putString(Constants.PREF_DOWNLOAD_LOCATION, defaultStorage) }
+                defaultStorage
+            }
+        }
+        set(value) {
+            sharedPreferences.edit {
+                with(File(value)) {
+                    parentFile?.run {
+                        if (exists() && isDirectory) putString(Constants.PREF_DOWNLOAD_LOCATION, value)
+                    }
```

Comment:
Same as above.

Reply:
Now that I think about it, this check is redundant, since it is the same found in the `get()`, I will remove it.